test/e2e/issuer.ts

```diff
@@ -128,7 +158,103 @@ export async function testE2E(issuerName: string, mTLS?: MTLSConfiguration): Pro
 		response.headers.get('Content-Type') === MediaType.PRIVATE_TOKEN_RESPONSE
 	);
 }
+
+async function testArbitraryBatchedRequest(issuerName: string, nTokens: number, mTLS?: MTLSConfiguration) {
+	// Create an array of per-token client instances.
+	const clients: publicVerif.Client[] = [];
+	for (let i = 0; i < nTokens; i++) {
+		clients.push(new Client(BlindRSAMode.PSS));
+	}
+	// Create a separate master client for aggregating token requests.
+	const masterClient = new Client(BlindRSAMode.PSS);
+	// Create the origin instance.
+	const origin = new Origin(BlindRSAMode.PSS);
+
+	// Get issuer configuration.
+	const { publicKeyEnc: issuerPublicKeyEnc, publicKey: issuerPublicKey, url: issuerRequestURL } =
+		await getIssuerConfig(issuerName, mTLS);
+
+	const batchedTokensClient = new BatchedTokensClient();
+
+	// Create token challenges.
+	const tokChls = new Array<TokenChallenge>(nTokens);
+	for (let i = 0; i < nTokens; i++) {
+		const redemptionContext = crypto.getRandomValues(new Uint8Array(32));
+		tokChls[i] = origin.createTokenChallenge(issuerName, redemptionContext);
+	}
+
+	// Create individual token requests using the per-token clients.
+	const tokReqs = new Array<arbitraryBatched.TokenRequest>(nTokens);
+	for (let i = 0; i < tokChls.length; i++) {
+		try {
+			const rawTokReq = await clients[i].createTokenRequest(tokChls[i], issuerPublicKeyEnc);
+			tokReqs[i] = new TokenRequest(rawTokReq);
+		} catch (err) {
+			throw err;
+		}
+	}
+	// Use the master client to batch the individual token requests.
+	const tokenRequests = batchedTokensClient.createTokenRequest(tokReqs);
+	const serializedRequest = tokenRequests.serialize();
+
+	// Send the batched token request to the issuer.
+	const proxyFetch = mTLS ? await fetchWithMTLS(mTLS) : fetch;
+	const response = await proxyFetch(issuerRequestURL, {
+		method: 'POST',
+		headers: {
+			'Content-Type': MediaType.ARBITRARY_BATCHED_TOKEN_REQUEST,
+			'Accept': MediaType.ARBITRARY_BATCHED_TOKEN_REQUEST,
+		},
+		body: serializedRequest.buffer as Buffer, // node-fetch expects a Node.js Buffer
+	});
+
+	if (!response.ok) {
+		throw new Error(`Issuer request failed: ${response.status} ${response.statusText}`);
+	}
+
+	// Deserialize the batched token response.
+	const respBuffer = new Uint8Array(await response.arrayBuffer());
+	const tokenResponse = BatchedTokenResponse.deserialize(respBuffer);
+
+	// Finalize each token response using the corresponding client instance.
+	let j = 0;
+	const tokens = new Array<Token | undefined>(nTokens);
+	for (const res of tokenResponse) {
+		if (res.tokenResponse === null) {
+			console.warn(`Token Response [${j}] is null; skipping finalization.`);
+			j++;
+			continue;
+		}
+		const r = publicVerif.TokenResponse.deserialize(res.tokenResponse);
+		try {
+			tokens[j] = await clients[j].finalize(r);
+		} catch (err) {
+			throw err;
+		}
+		j++;
+	}
+
+	// Verify each finalized token.
+	let isValid = true;
+	for (let i = 0; i < tokens.length; i++) {
```

Comment:
```suggestion
	for (let i = 0; i < tokens.length; i += 1) {
```

stylistic suggestion, that's what this codebase uses